semantic_inference/python/semantic_inference/models/wrappers.py

```diff
@@ -335,3 +335,55 @@ class OpenClipConfig(Config):
     def load(cls, filepath):
         """Load config from file."""
         return Config.load(cls, filepath)
+
+
+class Yolov11InstanceSegmenterWrapper(nn.Module):
+    """Yolov11 instance segmentation wrapper."""
+
+    def __init__(self, config):
+        """Load Yolov11 model."""
+        super().__init__()
+        from ultralytics import YOLO
+
+        self.config = config
+        self.model = YOLO(config.model_name)
+
+    def eval(self):
+        """
+        override eval to avoid issues with yolo model
+        """
+        self.model.model.eval()
+
+    @classmethod
+    def construct(cls, **kwargs):
+        """Load model from configuration dictionary."""
+        config = Yolov11InstanceSegmenterConfig()
+        config.update(kwargs)
+        return cls(config)
+
+    def forward(self, img):
+        """Segment image."""
+        result = self.model(img)[0]  # assume batch size 1
+        if result.masks is None:
+            return None, None, None, None
```

Comment:
(minor) I tend to put newlines between the end of a control-flow statement and the start of a new block, i.e., 

```
if result.masks is None:
    return None, None,  None, None

categories = result.boxes.cls
```
instead of
```
if result.masks is None:
    return None, None,  None, None
categories = result.boxes.cls
```

Ruff doesn't do this / it's not strictly necessary, so no need to change it if you don't want to